code/espurna/config/general.h

```diff
@@ -333,53 +373,86 @@ PROGMEM const char* const custom_reset_string[] = {
 // WIFI
 // -----------------------------------------------------------------------------
 
+#ifndef WIFI_CONNECT_TIMEOUT
 #define WIFI_CONNECT_TIMEOUT    60000               // Connecting timeout for WIFI in ms
+#endif
+
+#ifndef WIFI_RECONNECT_INTERVAL
 #define WIFI_RECONNECT_INTERVAL 180000              // If could not connect to WIFI, retry after this time in ms
+#endif
+
 #define WIFI_MAX_NETWORKS       5                   // Max number of WIFI connection configurations
+
+#ifndef WIFI_AP_MODE
 #define WIFI_AP_MODE            AP_MODE_ALONE
+#endif
+
+#ifndef WIFI_SLEEP_MODE
 #define WIFI_SLEEP_MODE         WIFI_NONE_SLEEP     // WIFI_NONE_SLEEP, WIFI_LIGHT_SLEEP or WIFI_MODEM_SLEEP
+#endif
+
+#ifndef WIFI_SCAN_NETWORKS
 #define WIFI_SCAN_NETWORKS      1                   // Perform a network scan before connecting
+#endif
 
 // Optional hardcoded configuration (up to 2 networks)
 #ifndef WIFI1_SSID
 #define WIFI1_SSID              ""
-#endif
+
 #ifndef WIFI1_PASS
 #define WIFI1_PASS              ""
 #endif
+
 #ifndef WIFI1_IP
 #define WIFI1_IP                ""
 #endif
+
 #ifndef WIFI1_GW
 #define WIFI1_GW                ""
 #endif
+
 #ifndef WIFI1_MASK
 #define WIFI1_MASK              ""
 #endif
+
 #ifndef WIFI1_DNS
 #define WIFI1_DNS               ""
 #endif
+
+#endif // WIFI1
```

Comment:
And this is where it got lost. If ssid is defined other values are undefined.
https://github.com/xoseperez/espurna/blob/fa20c47f2905c46628cbdde475e899ecc30368a3/code/espurna/wifi.ino#L174-L183

Reply:
well, it's kind of "by design"  - if you don't specify WIFI1 SSID, than rest of the WIFI1 settings are pointless. I can change back to "old" behavior, but I see no usecase of setting PASS or GW, without actually setting SSID.

Reply:
I am mostly talking about logic above. If WIFI1_SSID is defined, but nothing else is:
```
espurna/code/espurna/wifi.ino: In function 'void _wifiInject()':
espurna/code/espurna/wifi.ino:178:35: error: 'WIFI1_PASS' was not declared in this scope
setSetting("pass", 0, WIFI1_PASS);
^
espurna/code/espurna/wifi.ino:179:33: error: 'WIFI1_IP' was not declared in this scope
setSetting("ip", 0, WIFI1_IP);
^
espurna/code/espurna/wifi.ino:180:33: error: 'WIFI1_GW' was not declared in this scope
setSetting("gw", 0, WIFI1_GW);
^
code/espurna/wifi.ino:181:35: error: 'WIFI1_MASK' was not declared in this scope
setSetting("mask", 0, WIFI1_MASK);
^
code/espurna/wifi.ino:182:34: error: 'WIFI1_DNS' was not declared in this scope
setSetting("dns", 0, WIFI1_DNS);
```

Reply:
erm, right...
didn't see that in Travis (still shows success even if build fails).